86BoxManager/Exporter/SettingsExporter.cs

```diff
@@ -0,0 +1,354 @@
+﻿using Microsoft.Win32; 
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression; 
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace _86boxManager
+{
+    /// <summary>
+    /// 2020-11-23  Connor Hyde (starfrost013)
+    /// </summary>
+    public class SettingsExporter
+    {
+        public bool ZipUpRegFile { get; set; }
+        
+        // we probably need input validation support and result classes,
+        // but i already wrote too much code for this pull request
+        public bool ExportSettings()
+        {
+            SaveFileDialog SFD = new SaveFileDialog();
+            
+            // If the user has chosen to zip up the reg file, modify the dialog accordingly
+            if (!ZipUpRegFile)
+            {
+                SFD.Title = "Export to Registry File";
+                SFD.Filter = "Registry file (*.reg)|*.reg";
+            }
+            else
+            {
+                SFD.Title = "Export to Zipped Registry File";
+                SFD.Filter = "ZIP archive (*.zip)|*.zip";
+            }
+
+            SFD.ShowDialog();
+
+            if (SFD.FileName == "")
```

Comment:
I'm not sure what your intention here was, but you're missing a check for the DialogResult - if I cancel the dialog without saving anything, this still returns true as if export was successful.